### PR TITLE
Push sidebar under first full-width media row with push sidebar down option selected.

### PR DESCRIPTION
**Pull request by trackleft**
## Types of changes

**Arizona Quickstart** (install profile, custom modules, custom theme)
- Patch release changes
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

**Drupal core**
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

**Drupal contrib projects**
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by ejsamboy** [comment]
I ran a few tests, and the sidebar menu moves under the first full-width **Text with Background** feature when the "Push sidebar down below this element" option is selected. The sidebar seems to be working fine.

File: modules/custom/az_paragraphs/js/az_paragraphs_push_sidebar_down.es6.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        );
+        const contentRegionPosition = contentRegion.getBoundingClientRect();
+        const style =
+          sidebarPusher[0].currentStyle ||
+          window.getComputedStyle(sidebarPusher[0], "");
+        const bottomMargin = style.marginBottom;
+        const contentRegionTop = contentRegionPosition.top;
+        const sidebarPusherPosition = sidebarPusher[0].getBoundingClientRect();
+        const sidebarPusherBottom = sidebarPusherPosition.bottom;
+        const sidebarTopMargin = sidebarPusherBottom - contentRegionTop;
```
Comment (trackleft) [inline modules/custom/az_paragraphs/js/az_paragraphs_push_sidebar_down.es6.js:17]:
```suggestion
        const sidebarTopMargin = (sidebarPusherBottom - contentRegionTop) + bottomMargin;
```

File: modules/custom/az_paragraphs/js/az_paragraphs_push_sidebar_down.es6.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        document.documentElement.style.setProperty(
+          "--sidebar-top-margin",
+          `calc(${sidebarTopMargin}px + ${bottomMargin})`
```
Comment (trackleft) [inline modules/custom/az_paragraphs/js/az_paragraphs_push_sidebar_down.es6.js:20]:
```suggestion
          `${sidebarTopMargin}px`
```

File: modules/custom/az_paragraphs/src/Plugin/paragraphs/Behavior/AZPushSidebarDownBehavior.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+   * {@inheritdoc}
+   */
+  public function buildBehaviorForm(ParagraphInterface $paragraph, array &$form, FormStateInterface $form_state) {
+
+    // Get stored plugin configuration for this paragraph.
+    $config = $this->getSettings($paragraph);
+
+    $form['push_sidebar_down'] = [
+      '#title' => $this->t('Push sidebar down below this element.'),
+      '#type' => 'select',
```
Comment (trackleft) [inline modules/custom/az_paragraphs/src/Plugin/paragraphs/Behavior/AZPushSidebarDownBehavior.php:85]:
```suggestion
```

**Pull request closed without merging**